samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java

```diff
@@ -1959,6 +1961,29 @@ public boolean getHasAttachment(){
     return hasAttachment;
   }
 
+  public boolean getShowFeedbackLink() {
+    return (getActionString() == "takeAssessment"
+            || getActionString() == "takeAssessmentViaUrl"
+            || getActionString() == "previewAssessment"
+           ) && getNavigation() != "1"
+             && getPageContents().getIsNoParts();
+  }
+
+  public boolean getShowTableOfContentLink() {
+    return (getActionString() == "takeAssessment"
+            || getActionString() == "takeAssessmentViaUrl"
+            || getActionString() == "previewAssessment"
+           ) && getNavigation() != "1";
+  }
+
+  public boolean getShowReturnToAssessmentLink() {
+    return getActionString() == "reviewAssessment" && isAnonymousLogin() && isToolHidden();
```

Comment:
StringUtils.equals(getActionString(), REVIEW_ASSESSMENT) the constant is probably already created.